agent/lm_agent/parsing/olicense.py

```diff
@@ -0,0 +1,137 @@
+"""
+Parser for OLicense
+"""
+import re
+from typing import Optional
+
+HOSTWORD = r"[a-zA-Z0-9-]+"
+HOSTNAME = rf"{HOSTWORD}(\.{HOSTWORD})*"
+INT = r"\d+"
+EXPIRATION_DATE = (
+    r"[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1]) (2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9]"
+)
+
+FEATURE_LINE = (
+    r"^\s+(?P<feature>\w+);"
+    r"\s+(?P<license_type>\w+);"
+    rf"\s+(?P<total>{INT});"
+    rf"\s+(?P<expiration_date>{EXPIRATION_DATE});"
+)
+
+IN_USE_LINE = rf"^\s+(?P<in_use>{INT})\s+FloatsLockedBy:$"
+
+USAGE_LINE = rf"^\s+(?P<user>\S+)@(?P<lead_host>{HOSTNAME})\s+#{INT}$"
+
+RX_FEATURE = re.compile(FEATURE_LINE)
+RX_IN_USE = re.compile(IN_USE_LINE)
+RX_USAGE = re.compile(USAGE_LINE)
+
+
+def parse_feature_line(line: str) -> Optional[dict]:
```

Comment:
I think this is a good point.

Reply:
100%, we could refactor all the parsers to use data classes.
Since this feature is high priority, I think we should keep it like this, and add a refactor task to the backlog.